cross_dock/services/excel_service.py

```
@@ -0,0 +1,194 @@
+
+    total_rows = len(data)
+    processed_rows = 0
+
+    for row_num, item in enumerate(data, start=2):
+        try:
+            brand = item['Бренд']
+            article = item['Артикул']
+            logger.info(f"Processing row {row_num-1}/{total_rows}: {brand}, {article}")
+        except KeyError as e:
+            logger.error(f"KeyError in row {row_num-1}: {e}. Item keys: {item.keys()}")
+            raise ValueError(f"Missing required column in row {row_num-1}: {e}. Available columns: {item.keys()}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)